include/camp/resource.hpp

```diff
@@ -144,6 +182,37 @@ namespace resources
       std::shared_ptr<ContextInterface> m_value;
     };
 
+    /*
+     * \brief Compares a generic Resource and a concrete Resource to see if they are equal. To find out, we
+     * first check to see if they are on the same platform. Then we call the operator == to compare the 
+     * type of the generic Resource (T) and the concrete one.
+     *
+     * \return True or false depending on comparison with m_value and t, assuming they are on the same platform.
```

Comment:
What does `m_value` refer to here?

Reply:
I'll change that, thanks

Reply:
In order to appease MSVC and SYCL compilers which were confused about by the dummy struct "Host2" created in the test, we decided to get rid of these specific-generic and generic-specific Resource comparisons... they can be added later perhaps...